File: ruby/samples/get_options.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR API KEY'
end
 
options_api = Intrinio::OptionsApi.new

symbol = "AAPL" # String | The option symbol, corresponding to the underlying security.

opts = { 
  type: "put", # String | The option contract type.
  strike: nil, # Float | The strike price of the option contract. This will return options contracts with strike price equal to this price.
  strike_greater_than: nil, # Float | The strike price of the option contract. This will return options contracts with strike prices greater than this price.
  strike_less_than: nil, # Float | The strike price of the option contract. This will return options contracts with strike prices less than this price.
  expiration: nil, # String | The expiration date of the option contract. This will return options contracts with expiration dates on this date.
  expiration_after: nil, # String | The expiration date of the option contract. This will return options contracts with expiration dates after this date.
  expiration_before: nil, # String | The expiration date of the option contract. This will return options contracts with expiration dates before this date.
  page_size: 100, # Float | The number of results to return
  next_page: nil # String | Gets the next page of data from a previous API call
}

begin
  result = options_api.get_options(symbol, opts)
  options = result.options

  puts "#{options.size} options found for #{symbol}!"

  options.each do |option|
    puts ""
    puts "ID:         #{option.id}"
    puts "Code:       #{option.code}"
    puts "Ticker:     #{option.ticker}"
    puts "Expiration: #{option.expiration}"
    puts "Strike:     #{option.strike}"
    puts "Type:       #{option.type}"
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling OptionsApi->get_options: #{e}"
end

File: java/JavaSDKExample/src/main/java/getOptions.java
package com.intrinio.javasdkexample;

import com.intrinio.invoker.ApiClient;
import com.intrinio.invoker.ApiException;
import com.intrinio.invoker.Configuration;
import com.intrinio.invoker.auth.*;
import com.intrinio.api.OptionsApi;
import com.intrinio.models.*;

import java.math.BigDecimal;
import java.util.List;

public class App
{
  public static void main( String[] args )
  {
    ApiClient defaultClient = Configuration.getDefaultApiClient();
    ApiKeyAuth auth = (ApiKeyAuth) defaultClient.getAuthentication("ApiKeyAuth");
    auth.setApiKey("YOUR API KEY");

    OptionsApi optionsApi = new OptionsApi();

    String symbol = "AAPL"; // String | The option symbol, corresponding to the underlying security.
    String type = "put"; // String | The option contract type.
    BigDecimal strike = null; // BigDecimal | The strike price of the option contract. This will return options contracts with strike price equal to this price.
    BigDecimal strikeGreaterThan = null; // BigDecimal | The strike price of the option contract. This will return options contracts with strike prices greater than this price.
    BigDecimal strikeLessThan = null; // BigDecimal | The strike price of the option contract. This will return options contracts with strike prices less than this price.
    String expiration = null; // String | The expiration date of the option contract. This will return options contracts with expiration dates on this date.
    String expirationAfter = null; // String | The expiration date of the option contract. This will return options contracts with expiration dates after this date.
    String expirationBefore = null; // String | The expiration date of the option contract. This will return options contracts with expiration dates before this date.
    BigDecimal pageSize = null; // BigDecimal | The number of results to return
    String nextPage = null; // String | Gets the next page of data from a previous API call

    try {
      ApiResponseOptions result = optionsApi.getOptions(symbol, type, strike, strikeGreaterThan, strikeLessThan, expiration, expirationAfter, expirationBefore, pageSize, nextPage);
      List<Option> options = result.getOptions();

      System.out.println(options.size() + " options found for " + symbol + "!");
      System.out.println();

      for (Option option: options) {
        System.out.println();
        System.out.println("ID:         " + option.getId());
        System.out.println("Code:       " + option.getCode());
        System.out.println("Ticker:     " + option.getTicker());
        System.out.println("Expiration: " + option.getExpiration());
        System.out.println("Strike:     " + option.getStrike());
        System.out.println("Type:       " + option.getType());
      }
    } catch (ApiException e) {
      System.err.println("Exception when calling OptionsApi#getOptions");
      e.printStackTrace();
    }
  }
}
